### PR TITLE
Use evm inputs for issuers

### PR DESCRIPTION
**Pull request by kattylucy**
Use the address input that we have for CFG transfers, that supports both EVM addresses and Centrifuge addresses

#2221 

### Approvals

- [ ] Dev
- [x ] Dev
- [ ] Designer
- [ x] Product

### Impact

Pool managers on Create pool page
Investor input for Centrifuge Chain on Investors tab
AO wallets on Access tab
Receiving address on Fees tab

### CODE REVIEW

File: centrifuge-app/src/pages/IssuerPool/Configuration/AddAddressInput.tsx [hunk 1]
```diff
@@ -19,19 +19,27 @@ export function AddAddressInput({
...
 
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    const newAddress = e.target.value
+    setAddress(newAddress)
+  }
+
+  function handleBlur(e: React.FocusEvent<HTMLInputElement>) {
+    const address = e.target.value
+    if (address && !exists) {
+      onAdd(addressToHex(address))
```
Comment (onnovisser) [inline centrifuge-app/src/pages/IssuerPool/Configuration/AddAddressInput.tsx:35]: Is this the desired behaviour? If so, it should check that it's a valid address (check if `truncated` is truthy)

File: fabric/src/components/TextInput/index.tsx [hunk 2]
```diff
@@ -252,7 +252,12 @@ export function TextAreaInput({
 }
 
 export type AddressInputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value'> &
-  Omit<InputUnitProps, 'inputElement'>
+  Omit<InputUnitProps, 'inputElement'> & {
+    value?: string
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
+    onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void
```
Comment (sophialittlejohn) [inline fabric/src/components/TextInput/index.tsx:258]: I don't think you have to explicitly declare these three types, they should already be included in the type `React.InputHTMLAttributes<HTMLInputElement`

**Review by sophialittlejohn — COMMENTED** [review]
Great job! 

Can you please make sure the add address button is vertically centered? And the same for the pool managers?
Screenshot 2024-06-21 at 12 34 06

In the fees tab I noticed that when I use an EVM address it's saying "Invalid address", I think you might need to adjust the validation to make it work for EVM
Screenshot 2024-06-21 at 12 38 45

File: fabric/src/components/TextInput/index.tsx (hunk 2 above)
Comment (kattylucy) [reply fabric/src/components/TextInput/index.tsx:258]: Yes, you are right ! - thanks

**Comment by sophialittlejohn** [comment]
Nice work Katty! Just a few more things to clarify and then I think it's good to go!

I think we can now remove the dropdown next to the input since the network is now implied by the address, buuuut @hieronx will we still need it to differentiate between the different EVM chains for liquidity pools?
Screenshot 2024-06-25 at 13 48 46

I think we also need some kind of icon for the evm wallets here, maybe just the ethereum logo for now? I also tried submitting the transaction but it failed: `{"err":{"cannotLookup":null}}`. Maybe you're still missing the address conversion here? 
Screenshot 2024-06-25 at 13 51 11

**Comment by hieronx** [comment]
> I think we can now remove the dropdown next to the input since the network is now implied by the address, buuuut @hieronx will we still need it to differentiate between the different EVM chains for liquidity pools?

Yeah we still need to differentiate between EVM chains. For the Centrifuge network, the input could be either an EVM address or a Substrate/Centrifuge address. For all other networks, it's always an EVM address.

**Comment by kattylucy** [comment]
_Yeah we still need to differentiate between EVM chains._

How do we currently differentiate the two?

**Comment by hieronx** [comment]
@kattylucy that was in response to Sophia's question. The dropdown is for selecting the network

**Comment by sophialittlejohn** [comment]
Almost there, typing doesn't work anymore on this input

Screenshot 2024-07-18 at 17 24 54

File: centrifuge-app/src/pages/IssuerPool/Access/AssetOriginators.tsx [hunk 3]
```diff
@@ -156,6 +158,15 @@ function AOForm({
...
   } = useCentrifugeConsts()
 
+  React.useEffect(() => {
+    const fetchChainId = async () => {
+      const id = await cent.getChainId()
+      setChainId(id)
+    }
+
+    fetchChainId()
+  }, [cent])
```
Comment (sophialittlejohn) [inline centrifuge-app/src/pages/IssuerPool/Access/AssetOriginators.tsx:168]: this would make a good hook

**Review by sophialittlejohn — APPROVED** [review]
Left one more comment to clean up a bit but already looks good to me 🔥 nice work pushing this one through!

File: centrifuge-app/src/pages/IssuerPool/Access/AssetOriginators.tsx [hunk 4]
```diff
@@ -156,6 +158,15 @@ function AOForm({
     transferAllowlist: { receiverDeposit },
   } = useCentrifugeConsts()
 
+  React.useEffect(() => {
+    const fetchChainId = async () => {
+      const id = await cent.getChainId()
```
Comment (onnovisser) [inline centrifuge-app/src/pages/IssuerPool/Access/AssetOriginators.tsx:163]: this is the wrong chain id. you'll need to use `useCentEvmChainId()`

File: fabric/src/components/TextInput/index.tsx [hunk 5]
```diff
@@ -262,40 +265,47 @@ export function AddressInput({
...
   onBlur,
   onChange,
+  value = '',
+  clearIcon,
   ...inputProps
 }: AddressInputProps) {
   const defaultId = React.useId()
   id ??= defaultId
 
+  const [inputVal, setInputVal] = React.useState<string>(value)
```
Comment (onnovisser) [inline fabric/src/components/TextInput/index.tsx:275]: the state doesn't seem necessary. the effect can use the `value` prop. using state here can lead to a bug when the value is changed outside the component, causing the effect not to run

**Review by onnovisser — APPROVED** [review]

File: centrifuge-js/src/utils/index.ts [hunk 6]
```diff
@@ -137,6 +137,8 @@ export function computeMultisig(multisig: Multisig): ComputedMultisig {
 }
 
 export function evmToSubstrateAddress(address: string, chainId: number) {
+  if (!chainId) return
```
Comment (onnovisser) [inline centrifuge-js/src/utils/index.ts:140]: better to throw in that case

**Pull request merged**